libraries/primitive_consul_lock.rb

```diff
@@ -85,7 +86,8 @@ def wait_until(action, opts = {})
       if success
         Chef::Log.info "#{action.to_s.capitalize}ed the lock #{path}"
       else
-        Chef::Log.warn "Will ignore errors and since we've reached #{opts[:max_failures]} errors"
+        Chef::Log.warn "Will ignore errors and since we've reached #{max_failures} errors"
+        raise "Max attempt #{max_failures} reached" if max_failures && !opts[:continue]
```

Comment:
OPT: For `max_failures` you now have a global option (`@options`) but you didn't do the same for `continue` is it expected? Maybe add a global option.

**Mandatory:** the implementation used to not raise, now it will fail except if the option is specified, this is a breaking change, for a not super critical feature.
I think you should either invert the flag `fail_on_max_failures`  (default `false`) or configure a default value of `true` to your new `continue` option.

OPT: document this new feature somehow

NIT: `continue` is not super explicit `continue_on_failure` would longer but be more explicit.

Tip: If you add a global option you could define a local variable as you did for `max_failures`, it would also give you a trivial opportunity to define the default value to `true`